lib/azure/armrest/storage_account_service.rb

```diff
@@ -292,7 +292,21 @@ def get_private_images(storage_accounts)
             next unless File.extname(blob.name).casecmp('.vhd') == 0
             next unless blob.properties.lease_state.casecmp('available') == 0
 
-            blob_properties = storage_account.blob_properties(blob.container, blob.name, key)
+            max_retries = 0
+
+            begin
+              blob_properties = storage_account.blob_properties(blob.container, blob.name, key)
+            rescue Errno::ECONNREFUSED, Azure::Armrest::TimeoutError => err
+              if max_retries < 3
+                max_retries += 1
+                sleep 1
+                retry
+              else
+                warn "WARNING: Unable to collect blob properties for #{blob.name}/#{blob.container}: #{err}"
```

Comment:
Yeah, I guess I should use the actual log there. We may have to revisit logging in the near future since there has been an enhancement to the rest-client gem, but that's another story.